src/Raven.Server/Documents/Queries/Parse/SimpleQueryParser.cs

```diff
@@ -20,7 +15,7 @@ public static IEnumerable<string> GetTermValuesForField(IndexQueryServerSide que
             if (string.IsNullOrWhiteSpace(query.Query))
                 yield break;
 
-            var q = QueryBuilder.BuildQuery(query.Query, query.DefaultOperator, query.DefaultField, QueryAnalyzer);
+            var q = QueryBuilder.BuildQuery(query.Parsed, query.Fields.Where, QueryAnalyzer);
```

Comment:
I don't follow. We don't parse the query again. Here we build Lucene query based on already parsed tokens.